### PR TITLE
Remove all YAML references

### PR DESCRIPTION
**Pull request by lcobucci**
@guilhermeblanco I think this covers everything except the documentation files, should we also change them?

### CODE REVIEW

**Comment by guilhermeblanco** [comment]
@lcobucci Documentation will be revisited once we get close to an alpha release. For now we're safe to experiment/implement stuff.

File: UPGRADE.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Upgrade to 3.0
 
+## BC Break: Removed ``YAML`` mapping drivers.
+
+If your code relies on ``YamlDriver``  or ``SimpleYamlDriver``, you should change to
```
Comment (afoeder) [inline UPGRADE.md]: shouldn't that rather be 

> _must_ change to 

?

**Comment by TomasVotruba** [comment]
I love the cleanup! GJ

**Comment by Ocramius** [comment]
Since everyone keeps asking, and a lot of YAML afficionados keep poking me on other channels, I'm clarifying the reasoning for this change.

 1. @guilhermeblanco is rewriting the internals of the mapping component, both for performance and for type-safety/ease of usage
 2. due to the mapping internal changes, mapping drivers would have required constant re-adapting. We simply don't have the throughput to deal with that, so we decided to get rid of some code instead.
 3. we are aiming for less mapping drivers to maintain ourselves. If YAML is such a big deal, then somebody may write a driver for it once the metadata changes are complete. It would not live in the `Doctrine\` namespace though.
 4. we aim at unifying mapping drivers via https://webmozart.io/blog/2014/10/24/defining-php-annotations-in-xml/ (not yet there, but that's the aim)
 5. Over the years, we got hundreds of help requests due to a missing key, a lacking indentation level, etc. These are not important problems, they are just noise, and given the limited amount of time that all of us have, we identified this as the actual problem. Getting rid of an "unfit-for-work" format seemed like a better decision.

If you want to use the `develop` branch, consider using the `orm:convert-mapping` command to translate your YML definitions into XML for now.

**Comment by mickaelandrieu** [comment]
@Ocramius thanks for the complete explanation.

So, to sum up what format should be used: php annotations (please no :( ) or xml files ? (Yay!).

Mickaël

**Comment by Ocramius** [comment]
@michael-lavaveshkul annotations are good for in-house projects, please use XML for anything reusable/published.

**Comment by GeeH** [comment]
@Ocramius will you still be supporting CSV please?

**Comment by Ocramius** [comment]
> @Ocramius will you still be supporting CSV please?

I hear that IDE support is poor, so we won't do it.

**Comment by GeeH** [comment]
Touchè :trollface:

**Comment by TomasVotruba** [comment]
@ocramius Thank you, explanation is always great.

**Comment by guilhermeblanco** [comment]
I'm gonna expand the work that is being done on develop branch, so any adventurous person won't shoot me on street.

Quote strategy is mostly gone, since it's not up to the ORM handle discrepancies on user's database mapping. If you define your columns and provide the column name all lowercase, it's not up to us fix it on your Oracle driver and make it uppercased. Instead of handling this for the user, which makes our code harder to maintain and also introduce serious WTF bugs (anyone interested to see some, read this method: https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Persisters/Entity/JoinedSubclassPersister.php#L233 ), we decided to leverage an easier way for users to define their own NamingStrategy. People are welcome to implement their own, solving their own casing problems.

Right now, I commented out *ALL* support for Embeddables. Every tiny bit I was changing was revealing random bugs, which hint a bad implementation under the hood. We'll bring it back before first alpha in a new, deeply integrated support, but for now, no go.

All 300's unit tests that are failing are due to quoting being now enforced everywhere. I dropped the backtick support and every single column, table, schema, etc is now quoted. I have yet to find a solution that makes quote comparisons proper and work across all drivers when running the unit tests.

**Comment by BallisticPain** [comment]
I've never been a fan of XML files, but I understand where your decision is coming from. I haven't used the YAML mapping since the introduction of annotations. I'll keep in mind if I ever release a library it's preferred to be in XML... why is that though?

I know previous reasons for XML vs YAML is validation, but YAML now can be validated I thought... so just curious.

Thanks for all of your hard efforts!

**Comment by Ocramius** [comment]
> YAML now can be validated I thought... so just curious.

We'd have to write the validation logic for it, and that is just overhead.

**Comment by theofidry** [comment]
@guilhermeblanco nobody is gonna shot you in the street because you made a sensible decisions ;)

I really like YAML so I belongs to the person disappointed by this decision, but I perfectly understand the rationale so thanks very much for the explanation and for working so hard for Doctrine 3 👍 👌

**Comment by BallisticPain** [comment]
@Ocramius 
Fully understand, I'd rather the focus be on what y'all do best than the ancillary parts. Those that really want the YAML support can create an additional library to add that support with proper focus on the YAML.

Now why is it that XML is the standard requested for third-party available libraries? Is it because it's so verbose?

**Comment by theofidry** [comment]
@BallisticPain XML it is a tool of choice for configuration as you can validate your config against an XSD schema. For example when using PHPUnit, if you have:

```xml
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/5.2/phpunit.xsd">
```

You'll be able to have syntax highlighting and auto-completion when writing your PHPUnit file.

**Comment by Ocramius** [comment]
@BallisticPain what @theofidry said. In addition to that, should we change our schema to remove configs that we don't support, or alter how they are to be set up, you'd get immediate feedback from the XSD validation (which would fail, highlighting the issue).

**Comment by BallisticPain** [comment]
@theofidry @Ocramius 
Thanks! That makes a lot of sense. I just usually try to stay away from XML is the reason I'm asking. I've used it in almost every project, but I avoid it for a while before jumping in ... :-P

**Comment by JarJak** [comment]
I'd love if someone would still maintain YAML driver.

**Comment by Ocramius** [comment]
We're having major issues in keeping the most important bits maintained and supported (as in helping people out): either somebody builds a repository for YAML support (and maintains that outside of the doctrine organization) or it is not going to happen for now.

**Comment by rdohms** [comment]
I represent the repressed people of Yaml and would like to say I stand strong against this carnage of configuration of the yaml kind. YAML IS ALSO TEXT!

Actually, no, kill it with fire.

**Comment by mickaelandrieu** [comment]
Even if I'm sad (was one of YAML users), I do understand your choices: time and motivation aren't unlimited :)

**Comment by Ocramius** [comment]
Since I'm lazy, here's a longer explanation of why we're dropping the thing: https://github.com/doctrine/DoctrineBundle/issues/776#issuecomment-363812362

**Comment by zquintana** [comment]
Is Yaml support being dropped? Just visited the documentation today to find it's no longer there for associations.

**Comment by afoeder** [comment]
yes @zquintana see this PR and the immediate last comment prior to yours.

**Comment by Majkl578** [comment]
Looks like docs is being build from master, that's not a good idea since master now points to version 3.0-dev with BC breaks and many changes.

@ocramius can we somehow set the build process to use 2.7 branch instead?

**Comment by Ocramius** [comment]
@Majkl578 I think we should rather have proper versioning of the docs, no? I think @mikeSimonson and @malarzm were working on that.

**Comment by iamromeo** [comment]
@Ocramius I am sorry but what will happen to Symfony projects which used YAML files almost exclusively and the primary driver? Should we not be updating Doctrine anymore? Is there a converter to at least move from YAML to XML and/or Annotations?

File: lib/Doctrine/ORM/Tools/Setup.php [hunk 2]
```diff
@@ -19,13 +19,12 @@
 
 namespace Doctrine\ORM\Tools;
 
-use Doctrine\Common\ClassLoader;
+use Doctrine\Common\Cache\ArrayCache;
```
Comment (Quix0r) [inline lib/Doctrine/ORM/Tools/Setup.php]: This has been swapped. All fine?

File: lib/Doctrine/ORM/Tools/ConvertDoctrine1Schema.php [hunk 3]
```diff
@@ -1,343 +0,0 @@
-<?php
```
Comment (Quix0r) [inline lib/Doctrine/ORM/Tools/ConvertDoctrine1Schema.php]: Lesser code to take care of.

File: lib/Doctrine/ORM/Tools/Export/Driver/YamlExporter.php [hunk 4]
```diff
@@ -1,256 +0,0 @@
-<?php
```
Comment (Quix0r) [inline lib/Doctrine/ORM/Tools/Export/Driver/YamlExporter.php]: Lesser code to take care of.

File: tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3711Test.php [hunk 5]
```diff
@@ -1,29 +0,0 @@
-<?php
```
Comment (Quix0r) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3711Test.php]: Lesser code to take care of.

File: tests/Doctrine/Tests/ORM/Mapping/Symfony/YamlDriverTest.php [hunk 6]
```diff
@@ -1,40 +0,0 @@
-<?php
```
Comment (Quix0r) [inline tests/Doctrine/Tests/ORM/Mapping/Symfony/YamlDriverTest.php]: Lesser code to take care of.

File: tests/Doctrine/Tests/ORM/Mapping/YamlMappingDriverTest.php [hunk 7]
```diff
@@ -1,88 +0,0 @@
-<?php
```
Comment (Quix0r) [inline tests/Doctrine/Tests/ORM/Mapping/YamlMappingDriverTest.php]: Lesser code to take care of.

File: tests/Doctrine/Tests/ORM/Mapping/yaml/Doctrine.Tests.Models.CMS.CmsAddress.dcm.yml [hunk 8]
```diff
@@ -1,64 +0,0 @@
-Doctrine\Tests\Models\CMS\CmsAddress:
```
Comment (Quix0r) [inline tests/Doctrine/Tests/ORM/Mapping/yaml/Doctrine.Tests.Models.CMS.CmsAddress.dcm.yml]: Lesser code to take care of. And all the other `.yml` files as well.

File: tests/Doctrine/Tests/ORM/Tools/ConvertDoctrine1SchemaTest.php [hunk 9]
```diff
@@ -1,111 +0,0 @@
-<?php
```
Comment (Quix0r) [inline tests/Doctrine/Tests/ORM/Tools/ConvertDoctrine1SchemaTest.php]: Lesser code to take care of.

File: tests/Doctrine/Tests/ORM/Tools/Console/Command/ConvertDoctrine1SchemaCommandTest.php [hunk 10]
```diff
@@ -1,25 +0,0 @@
-<?php
```
Comment (Quix0r) [inline tests/Doctrine/Tests/ORM/Tools/Console/Command/ConvertDoctrine1SchemaCommandTest.php]: Lesser code to take care of.

File: tests/Doctrine/Tests/ORM/Tools/Export/AbstractClassMetadataExporterTest.php [hunk 11]
```diff
@@ -380,13 +378,6 @@ public function testCascadeAllCollapsed()
             self::assertEquals(1, count($nodes));
 
             self::assertEquals('cascade-all', $nodes[0]->getName());
-        } else if ($type == 'yaml') {
```
Comment (Quix0r) [inline tests/Doctrine/Tests/ORM/Tools/Export/AbstractClassMetadataExporterTest.php]: One static code block lesser.

File: tests/Doctrine/Tests/ORM/Tools/Export/YamlClassMetadataExporterTest.php [hunk 12]
```diff
@@ -1,44 +0,0 @@
-<?php
```
Comment (Quix0r) [inline tests/Doctrine/Tests/ORM/Tools/Export/YamlClassMetadataExporterTest.php]: Lesser code to take care of.

**Review by Quix0r — COMMENTED** [review]
In the end, nice cleanup. I didn't saw any non-yaml related stuff being accidentally removed.

**Comment by SergeyPodgornyy** [comment]
Peace of shit :hankey: your XML... everyone who like XML more then YML are freaks...

**XML sucks! Not as much as annotations, but still the same shit** :hankey:

**Comment by theofidry** [comment]
@SergeyPodgornyy unless you are willing to set up to provide proper support for YAML why are you being disrespectful and offensive instead? OSS is toxic enough as it is, no need to spread more of it

**Comment by Quix0r** [comment]
As I pointed out in my comments, lesser static but more driver/provider-based code is more flexible. It is good that this project heads that way. Still I miss real annotation support in PHP itself like it has been done in Java. `@Override` seem to be a superfluous annotation but it helps a lot when it comes to implementing against interfaces or super classes (inherited classes). Okay, wrong tracker, just my feelings about them. :-)

**Pull request merged**